lib/src/widgets/modal_widget.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:cake_wallet/utils/responsive_layout_util.dart';
+import 'package:flutter/material.dart';
+import 'package:cake_wallet/src/widgets/alert_background.dart';
+import 'package:cake_wallet/src/widgets/alert_close_button.dart';
+
+class ModalWidget extends StatelessWidget {
+  ModalWidget({required this.children, this.title});
+
+  final List<Widget> children;
+  final String? title;
+
+  @override
+  Widget build(BuildContext context) {
+    final double padding = 24;
+
+    final mq = MediaQuery.of(context);
+    final bottom = mq.viewInsets.bottom;
+    final height = mq.size.height - bottom;
+    final screenCenter = height / 2;
+
+    double closeButtonBottom = 60;
+    double containerHeight = height * 0.65;
+    if (bottom > 0) {
+      // increase a bit or it gets too squished in the top
+      containerHeight = height * 0.75;
+
+      final containerCenter = containerHeight / 2;
+      final containerBottom = screenCenter - containerCenter;
+
+      final hasTitle = this.title == null || this.title!.isEmpty;
+
+      // position the close button right below the search container
+      closeButtonBottom = closeButtonBottom -
+          containerBottom +
+          (hasTitle ? padding : padding / 1.5);
```

Comment:
when you check the bool condition, please verify that this line is working as intended